README.md

```diff
@@ -0,0 +1,69 @@
+* Open the [Google Cloud Build GitHub app](https://github.com/apps/google-cloud-build), click Install, and follow the setup instructions.  Select the `hello-kubernetes` repository you just created.
+* Back in your forked repo in GitHub, click to open `main_test.go`.  You will see some Go tests which make sure that the greeting matches the name provided.
+* Click the pencil icon in the top right-hand corner of the screen.
+* Edit one of the greetings so that the greeting does not match the name.
+* At the bottom of the screen, choose "Create a new branch and start a pull request".  Click "Propose file change".
+* On the page which opens, click "Create pull request".
+* You should see a message: "All checks have failed".  If you wish, you can see the build log by clicking the "Details" button and then selecting the link at the bottom of the screen which says "View more details on Google Cloud Build".
+* Next, choose the "Files changed" tab, and click the pencil icon again.
+* Make another edit so that the greetings now match the names.
+* Click the "Commit changes" button at the bottom of the screen.
+* Go back to the "Conversation" tab.
+* You should see that tests pass, and a green check is now returned.
+
+## Automated deployment
+
+Google Cloud Build can be used to deploy to Kubernetes automatically whenever you merge to a particular Git branch or tag.  Here, we're going to deploy to Google Kubernetes Engine (GKE) whenever a merge to `master` occurs.
+
+Before we start, we need to enable the GKE API, create a Kubernetes cluster for testing, and grant deployment permission to the Cloud Build service account.  To do this, copy and paste the following lines into your terminal:
+
+```sh
+gcloud services enable container.googleapis.com
+gcloud container clusters create my-cluster --zone=us-central1-f
+export PROJECT=$(gcloud info --format='value(config.project)')
+export PROJECT_NUMBER=$(gcloud projects describe $PROJECT --format 'value(projectNumber)')
+export CB_SA_EMAIL=$PROJECT_NUMBER@cloudbuild.gserviceaccount.com
+gcloud projects add-iam-policy-binding $PROJECT --member=serviceAccount:$CB_SA_EMAIL --role='roles/container.developer'  
+```
+
+Then:
+
+* Visit the [Build Triggers page](https://console.cloud.google.com/cloud-build/triggers) in the Cloud Console.
+* Click "Create Trigger".
+* Choose source GitHub and click Continue.  Follow the authorization flow to grant Google access to your code in GitHub.
+* Choose the `hello-kubernetes` repo, check the consent box, and click Continue.
+* Choose a name for your trigger, e.g. "Deploy to Kubernetes"
+* Under Branch (regex), enter `master`.
+* Under Build configuration, choose "cloudbuild.yaml", and enter filename `deploy.yaml`.
+* Click Save.
+* Now, return to GitHub, and click `main.go` to open the file.
+* Click the pencil icon on the top right-hand corner to edit.
+* On line 16, change the string inside `Greet("...")` to be your name.
```

Comment:
Now I broke the test!
Suggest:

- up above, change the test to your name
- then to fix the broken test in the PR, change main.go to your name
- then when you come back here, come back to the PR, the test has passed, so merge into Master

Reply:
Discussed in person, thanks!